### PR TITLE
fix(openai): include responses instructions in captured prompt

### PR DESCRIPTION
**Pull request by D-Joey-G**
Corrects behaviour such that arguments passed to the `instructions` parameter of the OpenAI responses API are treated like `'role': 'system'` arguments in the chat.completions API. (see https://github.com/langfuse/langfuse/issues/9775 or https://github.com/langfuse/langfuse/issues/10143 or https://github.com/langfuse/langfuse/issues/8763)

Adds tests to validate behaviour.

**Disclaimer**: Experimental PR review
---

<h3>Greptile Summary</h3>

This PR fixes the OpenAI Responses API integration so that the `instructions` parameter is captured in the Langfuse prompt alongside `input`, mirroring how `role: system` messages are treated in the Chat Completions API. It adds the `_extract_responses_prompt` helper and updates the existing streaming integration test and adds new unit tests.

**Key changes:**
- New `_extract_responses_prompt` function merges `instructions` + `input` into a unified `[{role: system}, {role: user}]` message list (or a dict fallback) before logging to Langfuse
- The function correctly normalises OpenAI SDK `NotGiven` sentinel values to `None`
- `_get_langfuse_data_from_kwargs` now delegates to `_extract_responses_prompt` instead of reading `input` directly for the `Responses`/`AsyncResponses` objects
- New parametrised unit test file (`test_openai_prompt_extraction.py`) covers all documented input combinations
- The streaming integration test is updated to assert the new merged prompt format

**Issues noted:**
- When `input` is a list that already contains a `{"role": "system", ...}` entry and `instructions` is also provided, the captured Langfuse prompt will have two system messages — the new one prepended unconditionally. This could mislead prompt replay or evaluation tools.
- No non-streaming integration test exercises the `instructions` + `input` path; only the streaming case is covered at the integration level.

<h3>Confidence Score: 4/5</h3>

- This PR is safe to merge; the fix is correct for all common usage patterns and doesn't alter what is sent to OpenAI.
- The core logic is sound and well-tested for all documented input shapes. The main concern is the edge case where a caller already includes a system message inside the `input` list — `instructions` will be prepended unconditionally, producing duplicate system entries in the captured Langfuse prompt. This doesn't affect the actual API call (only what Langfuse records) and is unlikely in practice, but it's a correctness issue worth addressing. A missing non-streaming integration test is a minor coverage gap. No regressions are expected for existing callers.
- Pay close attention to `langfuse/openai.py` lines 271-272 — the unconditional prepend of the system message when `input` is a list.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| langfuse/openai.py | Adds `_extract_responses_prompt` function that merges `instructions` and `input` into a unified prompt structure for the Responses API; logic is correct but has an untested edge case where a list `input` already containing a system message would get a duplicate prepended system entry. |
| tests/test_openai.py | Updates `test_response_api_streaming` to assert the new merged `[system, user]` input format; no new non-streaming integration test covers the `instructions` + `input` path. |
| tests/test_openai_prompt_extraction.py | New unit test file covering all documented cases for `_extract_responses_prompt`, including `NOT_GIVEN` sentinel values and string/list inputs; well-structured and complete for the happy paths. |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant User
    participant LangfuseWrapper as Langfuse Wrapper (_wrap)
    participant ExtractPrompt as _extract_responses_prompt
    participant Langfuse
    participant OpenAI as OpenAI Responses API

    User->>LangfuseWrapper: responses.create(model, instructions, input, ...)
    LangfuseWrapper->>ExtractPrompt: kwargs (instructions, input)
    alt instructions is None
        ExtractPrompt-->>LangfuseWrapper: input as-is
    else input is str
        ExtractPrompt-->>LangfuseWrapper: [{role:system, content:instructions}, {role:user, content:input}]
    else input is list
        ExtractPrompt-->>LangfuseWrapper: [{role:system, content:instructions}, ...input]
    else input is None
        ExtractPrompt-->>LangfuseWrapper: {instructions: instructions}
    end
    LangfuseWrapper->>Langfuse: start_observation(input=merged_prompt)
    LangfuseWrapper->>OpenAI: original call (unchanged kwargs)
    OpenAI-->>LangfuseWrapper: response
    LangfuseWrapper->>Langfuse: update generation (output, usage, model)
    LangfuseWrapper-->>User: response
```

<sub>Last reviewed commit: 4c8353f</sub>

> Greptile also left **1 inline comment** on this PR.

<sub>(2/5) Greptile learns from your feedback when you react with thumbs up/down!</sub>

### CODE REVIEW

File: langfuse/openai.py [hunk 1]
```diff
@@ -246,6 +246,34 @@ def wrapper(wrapped: Any, instance: Any, args: Any, kwargs: Any) -> Any:
...
+        return {"instructions": instructions}
+
+    if isinstance(input_value, str):
+        return [
+            {"role": "system", "content": instructions},
+            {"role": "user", "content": input_value},
+        ]
+
+    if isinstance(input_value, list):
+        return [{"role": "system", "content": instructions}, *input_value]
```
[the remark this answers is not in the document]
Comment (D-Joey-G) [reply langfuse/openai.py:272]: Multiple `developer/system` messages isn't uncommon, so that concern doesn't seem important.

I believe that it is safe/correct to put the `instructions` arg first in what we trace, as earlier docs for the Responses API said the parameter `inserts a system (or developer) message as the first item in the model's context.`

That said, the docs now say:
```
A system (or developer) message inserted into the model's context.

When using along with previous_response_id, the instructions from a previous response will not be carried over to the next response. This makes it simple to swap out system (or developer) messages in new responses.
```

That is admittedly less certain on that point.
